firmware.py

```diff
@@ -5,13 +5,55 @@
 
 app=Flask(__name__)
 app.config.from_object(__name__)
+app.config.update(dict(
+    DATABASE = os.path.join(app.root_path, 'firmware.db'),
+    SECRET_KEY='such secret much wow so key no guess muh'
+))
+
+
+def connect_db():
+    rv=sqlite3.connect(app.config['DATABASE'])
+    rv.row_factory = sqlite3.Row
+    return rv
+
+    
+def init_db():
+    db=get_db()
+    with app.open_resource('schema.sql',mode='r') as f:
+        db.cursor().executescript(f.read())
+    db.commit()
+
+
+@app.cli.command('initdb')
+def initdb_command():
+    init_db()
+    print("Firmware initialized database sucessfully!")
+def get_db():
+    if not hasattr(g,'sqlite_db'):
+        g.sqlite_db=connect_db()
+    return g.sqlite_db
+
+
+@app.teardown_appcontext
+def close_db(error):
+    if hasattr(g, 'sqlite_db'):
+        g.sqlite_db.close()
 
 
 @app.route('/')
 def show_home():
-   return render_template('index.html')
+    db = get_db()
+    cur = db.execute("select * from companies")
+    companies = cur.fetchall()
+    return render_template('index.html',companies=companies)
+
 
-   
-@app.route('/details')
-def show_details():
-   return render_template('details.html')
+@app.route('/details/<company_id>')
+def show_details(company_id):
+    db = get_db()
+    cur = db.execute("select user, review from reviews where company_id="+company_id+" order by id desc")
+    reviews = cur.fetchall()
+    db = get_db()
+    cur = db.execute("select * from companies where company_id="+company_id)
+    rows=cur.fetchall()
```

Comment:
`rows` doesn't tell me anything relevant about this variable. Use `company` instead and you can assign to it only the first records, which should be the only one. (You should check if there is a method which gives you only the first records and use that instead of `fetchall`.
 Also, you should check that the company id provided is a valid one and the company exists. In case that the company doesn't exist, you should show a message (or redirect to a 404 page)